web/src/components/core/Terminal.jsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from "react";
+import { Popup } from "~/components/core";
+
+export default function Terminal({ onCloseCallback }) {
+  // the popup is visible
+  const [isOpen, setIsOpen] = useState(true);
+
+  const close = () => {
+    setIsOpen(false);
+    if (onCloseCallback) onCloseCallback();
+  };
+
+  // embed the cockpit terminal into an iframe, see
+  // https://cockpit-project.org/guide/latest/embedding.html#embedding-components
+  // https://cockpit-project.org/guide/latest/api-terminal-html.html
+  return (
+    <Popup
+      isOpen={isOpen}
+      variant="large"
+      className="tallest"
+      aria-label="terminal popup"
+    >
+
+      <iframe data-testid="iframe-terminal" className="fill" src="/cockpit/@localhost/system/terminal.html" />
```

Comment:
I'd avoid the use of data-testid as much as possible. What is more, I'd use the `title` here for both, accessibility (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#accessibility_concerns) and testing (getByTitle).

Regarding a11n, I know that this particular iframe is alone and inside a popup that already describes what the user will find in it... but in any case...

Reply:
That `aria-label` is actually needed by an unit test, the `Modal` component complains that there is not title set. But the "Terminal" title is already included inside the `iframe` so it does not make sense to have it twice.